src/components/Analyze/charts/DoughnutChart.tsx

```diff
@@ -1,45 +1,52 @@
 import { AnalyzeContext } from '@/components/Analyze/AnalyzeContext'
-import { chartColors } from '@/Mock/Mock'
+import { ChartDataType } from '@/types/type'
 import { ResponsivePie } from '@nivo/pie'
 import { useContext, useEffect, useState } from 'react' // Import useEffect and useState
 
-interface ChartDataType {
-	id: string
-	label: string
-	value: number
-	color: string
-}
-
 export const DoughnutChart = () => {
-	const { doughnutChartData } = useContext(AnalyzeContext)
+	const { spendingData } = useContext(AnalyzeContext)
 	const [animatedData, setAnimatedData] = useState<ChartDataType[]>([])
 
+	const firstRow: ChartDataType[] = []
+	const secondRow: ChartDataType[] = []
+
+	spendingData?.map((each, idx) => {
+		if (idx <= spendingData.length / 2) firstRow.push(each.data)
+		else secondRow.push(each.data)
+	})
+
 	//차트가 변해야 애니메이션이 출력됨, default값을 넣고 0.05초뒤 필요한 값을 넣음
 	useEffect(() => {
-		if (!doughnutChartData) return
-		const initialData: ChartDataType[] = doughnutChartData.map((item) => ({
-			...item,
+		const initialData: ChartDataType[] = (spendingData || []).map((item) => ({
+			id: item.data.id,
+			label: item.data.label,
 			value: 0,
+			color: item.data.color,
 		}))
+
 		setAnimatedData(initialData)
 
 		const timeoutId = setTimeout(() => {
-			setAnimatedData(doughnutChartData)
+			setAnimatedData(
+				(spendingData || []).map((item) => ({
+					id: item.data.id,
+					label: item.data.label,
+					value: item.data.value,
+					color: item.data.color,
+				})),
+			)
 		}, 50)
-		return () => clearTimeout(timeoutId)
-	}, [doughnutChartData])
 
-	if (animatedData.length === 0) {
-		return null
-	}
+		return () => clearTimeout(timeoutId)
+	}, [spendingData])
 	return (
 		<ResponsivePie
 			data={animatedData}
 			margin={{ top: 10, right: 10, bottom: 50, left: 10 }}
 			innerRadius={0.5}
 			padAngle={0.7}
 			cornerRadius={3}
-			colors={chartColors}
+			colors={animatedData.flatMap((each) => [each.color])}
```

Comment:
여기서 `flatMap` 을 사용한 이유가 따로 있으신가요?
제가 처음 봐서 개념을 찾아보고 시뮬레이션을 돌려보니
1. `flatMap` 을 사용했을 때
: `[color1], [color2], [color3]` -> `[color1, color2, color3]`
2. `map` 을 사용했을 때
: `[color1, color2, color3]`
이렇게 되는 것 같은데 평탄화가 필요한지에 대해 고민해봤습니다.
그래서 아래 코드처럼 직접 추출해도 같은 결과이지 않을까 싶습니다.
```suggestion
			colors={animatedData.map((each) => each.color)}
```

Reply:
저도 신기해서 써봤던거긴 합니다.. ㅎㅎ 수정하려고 했는데 로직이 사라졌어욥

Reply:
어 요기 삭제 안 한거 아닌가요??

Reply:
아 그러네요 죄송합니다